proposals/filecoin-retrieval-in-ipfs.md

```diff
@@ -0,0 +1,163 @@
+
+5. We work with select pinning providers (eg Pinata) to verify we've provided the components for roundtrip pinning to Filecoin from IPFS
+
+6. We design the data transfer plugin API in concert with those who are looking at providing alternate mechanisms of retrieval (note: it's not clear to me if this should be a local plugin system or an HTTP API -- a local plugin can do any kind of custom HTTP communication, but is also harder to install and configure)
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+Done is when you can complete a roundtrip pinning / retrieval to Filecoin from IPFS and a spec exists for building future paid retrievals
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+- Well known pinning services begin offering Filecoin based pinning
+- Accelerator projects around new kinds of retrieval begin building plugins for paid retrieval
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+
+#### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+- A DHT change discussion is a big design discussion, and could take a long time or hit roadblocks. It also could involve breaking changes to the content routing API in Libp2p, which might cause a lot of headaches
+- It may turn out to be quite difficult to make smart decisions about when to use Filecoin retrieval vs regular Bitswap. The simple case where data is pinned ONLY in Filecoin and not IPFS is easy to decide on. When its in both, how we decide what to use is not totally clear.
+- The long term future of Filecoin retrieval protocol in general is dependent on improvements to proofs that enable faster unsealing or reliable systems for gauranteeing unsealed copies. (i.e. Filecoin Plus)
+
+#### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need?_
+
+See [Proposal: Free retrieval via IPFS ](https://github.com/protocol/web3-dev-team/pull/52). If all of that proposal's assumptions turn out to be true and it is possible to just have miners throw data in a bitswap accessible blockstore, it is probably the fastest path to baseline filecoin retrieval in IPFS. 
```

Comment:
If we want free retrieval without running a local Filecoin client then IMO that proposal seems like the thing to do.

However, the more complete solution to this problem requires a local Filecoin client and a proposal closer to this one